.gitlab-ci.yml

```diff
@@ -0,0 +1,25 @@
+image: archlinux/base
+
+# Change pip's cache directory to be inside the project directory since we can
+# only cache local items.
+variables:
+    PIP_CACHE_DIR: "$CI_PROJECT_DIR/.cache/pip"
+
+cache:
+  paths:
+    - .cache/pip
+
+before_script:
+   - pacman -Syu --needed --noconfirm python-pip git
+   - pip install pylint coverage
+   - pip install -r requirements.txt
```

Comment:
Why pip since we have everything in the repos?

Reply:
Well, not everything is in the repos. Also, we wouldn't want the project to break without knowing if something in Arch gets upgraded.

Reply:
Ah, sorry, looks like I missed some dependencies. But that's easily fixable. How are we deploying with missing dependencies?
If something updates and breaks archweb it would be good to know so that we can fix it, although I do see your point.

Reply:
For historical reasons we use pip, this might change in the future but indeed requires better / more up to date CI. Which is possible with gitlab, so it's probably a good idea to:
* package dependencies
* setup gitlab ci with packages and postgres/sqlite
* take it from there.